org.eclipse.xtext/src/org/eclipse/xtext/naming/QualifiedName.java

```diff
@@ -36,7 +36,7 @@
 
 	private QualifiedName lowerCase;
 
-	private static boolean USE_INTERNING = Boolean.getBoolean("xtext.qn.interning");
+	public static boolean USE_INTERNING = Boolean.getBoolean("xtext.qn.interning");
```

Comment:
I'm not a big fan of the undocumented public boolean flag here, but since the command line switch wasn't documented either, it's probably ok.

In other cases, where we added internal API via statics, we tried to hide it a little more, e.g. right now every code completion with the goal to show `QualifiedName.create` after the dot would also show `USE_INTERNING`.

Can we make this a protected field, such that the rare use case can be achieved by:
```
class QualifiedNameAccess extends QualifiedName {
  static void useInterning() {
    USE_INTERNING = true;
  }
}
```

Reply:
What about package private visibility without using inheritance?

Reply:
I had in mind that `QualifiedNameAccess` would not be in the Xtext codebase but part of the user code.

Reply:
I'd also be ok with a class next to QualifiedName, e.g. `QualifiedNameInterning` with public methods `enable()` and `disable()` and `USE_INTERNING` being package visible.

Reply:
Got it. I'm going to adjust the PR.